### ISSUE TITLE
Allow any xhr settings to be passed in via ajaxConfig

### ISSUE THREAD

**Issue by lukekarrys** [AmpersandJS/ampersand-sync#28]
Currently as said in https://github.com/AmpersandJS/ampersand-sync/issues/27 ajaxConfig can only be used to set a few different properties.

Would be nice for the other options `xhr` provides (such as `timeout`).

**Comment by chesles** [comment]
Is there a plan for this / anybody? In particular it would be really useful for testing to be able to set the `xhr` option (which is not documented right now, but there is a pending PR (Raynos/xhr#69) to add docs). Right now I'm doing something like this to make it work:

```js
model.fetch({
  xhrImplementation: function (opts, callback) {
    opts.xhr = new sinon.FakeXMLHttpRequest()
    return xhr(opts, callback)
  }
})
```

Is this blocked on/waiting on #32? Or is it something I could make a PR for now?

**Comment by naugtur** [comment]
xhr2 update + nodejs support landed in master, we can get back to this.

**Comment by naugtur** [comment]
bump. 
This looks to me like a potentially breaking change, but I don't know how `ajaxConfig` is used in the wild. 
If it's really breaking, let's consider adding it to 4.0 too.
@AmpersandJS/community-leaders can anybody comment on this?

**Comment by lukekarrys** [comment]
I think it could be breaking depending on the implementation, but it doesn't have to be. I do think it'd be nice if this landed in 4.0.

My original idea which I think would be non-breaking was to whitelist properties accepted by `xhr` and merge any of those found on `ajaxConfig` to `params`. Not sure if that would along with `request` now. Or just merge all properties found on ajaxConfig to `params`, but that seems like it might have potential side effects based on usage in the wild.

**Comment by naugtur** [comment]
ajaxConfig has `xhrFields` which is merged into the raw `XMLHttpRequest` object. It would have to be handled in a specific way.
Otherwise it seems to make sense to just copy it into options to let people override whatever we build from the regular input data. Whitelist seems heavy and not future compatible.

**Comment by lukekarrys** [comment]
> Otherwise it seems to make sense to just copy it into options to let people override whatever we build from the regular input data

:+1: That's what I expected to have happen back when I opened the issue. Agreed too on the whitelist, that was probably overthinking it.
